kernel/src/main/java/org/kframework/compile/checks/CheckAtt.java

```diff
@@ -108,15 +108,14 @@ private void checkLabel(Sentence sentence) {
     }
   }
 
-  private void checkMacro(Rule rule) {
-    if (rule.isMacro()) {
-      kem.registerCompilerWarning(
-          ExceptionType.FUTURE_ERROR,
-          errors,
-          "The attribute ["
-              + rule.att().getMacro().get()
-              + "] has been deprecated on rules. Use this label on syntax declarations instead.",
-          rule);
+  private void checkMacro(Sentence sentence) {
+    if (!(sentence instanceof Production) && sentence.isMacro()) {
+      errors.add(
+          KEMException.compilerError(
+              "The attribute ["
+                  + sentence.att().getMacro().get()
+                  + "] may only appear on syntax productions.",
+              sentence));
```

Comment:
I'd prefer if we us the general attribute whitelist infrastructure here instead of a manual error (i.e. update each `Key` in `Att.scala` to have `onlyon[Production]`).

Reply:
Very good point!